### PR TITLE
Moves deconstructing beds and chairs to right click

### PR DESCRIPTION
**Pull request by TheBonded**
## About The Pull Request

As part of my effort to both learn how to code DM and make all deconstruction happen on right click for consistency, I've started with the most prevalent thing I can think of: chairs (and beds too because they're in the same file). 

## Why It's Good For The Game

Picture this: you are deconstructing the coutroom, trying to get more iron because all the miners died and you want to print a fancy gun. You just finished removing some tables, by right clicking them with a wrench of course, and you come across a chair you want to deconstruct. So, holding your trusty wrench, you right click the chair, assuming that deconstructing the chair would be the same button as deconstructing those tables you just got done with. But no, that couldn't be further from the truth! You slam the wrench into the chair, smacking it with all your force, and making a massive fool of yourself. If only deconstruction was consistently on right click, you could have avoided this massive embarrassment.

Consistency good.

## Changelog

:cl:
qol: Chairs and beds are now deconstructed with right click, making them more consistent with other object deconstructions.
/:cl:

### CODE REVIEW

**Comment by TheBonded** [comment]
Draft pr until I can test this tomorrow afternoon

File: code/game/objects/structures/beds_chairs/bed.dm [hunk 1]
```diff
@@ -36,12 +36,12 @@
 /obj/structure/bed/attack_paw(mob/user, list/modifiers)
 	return attack_hand(user, modifiers)
 
-/obj/structure/bed/attackby(obj/item/W, mob/user, params)
-	if(W.tool_behaviour == TOOL_WRENCH && !(flags_1&NODECONSTRUCT_1))
-		W.play_tool_sound(src)
-		deconstruct(TRUE)
-	else
-		return ..()
+/obj/structure/bed/attackby_secondary(obj/item/weapon, mob/user, params)
```
Comment (Mothblocks) [inline code/game/objects/structures/beds_chairs/bed.dm:39]: Use `wrench_act_secondary`

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 2]
```diff
@@ -100,12 +100,13 @@
 		to_chat(user, "<span class='notice'> You cannot fit the shock kit onto the [name]!")
 
 
-/obj/structure/chair/wrench_act(mob/living/user, obj/item/I)
-	. = ..()
-	I.play_tool_sound(src)
-	deconstruct()
-	return TRUE
-
+/obj/structure/chair/attackby_secondary(obj/item/weapon, mob/user, params)
```
Comment (Mothblocks) [inline code/game/objects/structures/beds_chairs/chair.dm:103]: Use `wrench_act_secondary`

File: code/game/objects/structures/beds_chairs/bed.dm [hunk 3]
```diff
@@ -36,12 +36,13 @@
...
-		W.play_tool_sound(src)
-		deconstruct(TRUE)
-	else
-		return ..()
+/obj/structure/bed/wrench_act_secondary(mob/living/user, obj/item/weapon)
+	if(flags_1&NODECONSTRUCT_1)
+		return TRUE
+	..()
+	weapon.play_tool_sound(src)
+	deconstruct(TRUE)
```
Comment (Mothblocks) [inline code/game/objects/structures/beds_chairs/bed.dm:44]:
```suggestion
	deconstruct(disassembled = TRUE)
```

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 4]
```diff
@@ -100,12 +100,14 @@
...
-/obj/structure/chair/wrench_act(mob/living/user, obj/item/I)
-	. = ..()
-	I.play_tool_sound(src)
-	deconstruct()
+/obj/structure/chair/wrench_act_secondary(mob/living/user, obj/item/weapon)
+	if(flags_1&NODECONSTRUCT_1)
+		return TRUE
+	..()
+	weapon.play_tool_sound(src)
+	deconstruct(TRUE)
```
Comment (Mothblocks) [inline code/game/objects/structures/beds_chairs/chair.dm:108]:
```suggestion
	deconstruct(disassembled = TRUE)
```

**Pull request merged**